bam_addrprg.c

```diff
@@ -438,12 +438,17 @@ static bool init(const parsed_opts_t* opts, state_t** state_out) {
     return true;
 }
 
-static bool readgroupise(state_t* state)
+static bool readgroupise(parsed_opts_t *opts, state_t* state)
 {
     if (sam_hdr_write(state->output_file, state->output_header) != 0) {
         print_error_errno("addreplacerg", "[%s] Could not write header to output file", __func__);
         return false;
     }
+    char *idx_fn = NULL;
+    if (opts->ga.write_index && opts->output_name && strcmp(opts->output_name, "-") != 0) {
```

Comment:
`if (opts->ga.write_index)` should be enough, as `auto_index` checks internally for a null output file name and for `stdout`.